### DISCUSSION TITLE
Checking if tween is running?

### DISCUSSION THREAD

**Discussion by Debjoy** [jeffreylanters/unity-tweens#37] [Q&A] [answered]
How do you check if a particular tween you started is still running or not?

**Comment by jeffreylanters** [comment, marked as answer]
There isn't really a parameter that exposes the playing state. You can however add an event listener to its completion in order to check wether a tween is playing.

```cs
// Psuedo
var isPlaying = false;
target
  .TweenPositionX(1, 1)
  .SetOnStart(() => isPlaying = true)
  .SetOnComplete(() => isPlaying = false);
```

**Reply by Debjoy** [reply]
aah I see, wish I could though. but anyway thanks. I'll use similar logic to what you mentioned. Thanks

**Reply by jeffreylanters** [reply]
I'll try to take a look at it asap, to implement your request. It sounds like a good addition. I'll keep you posted!

**Reply by jeffreylanters** [reply]
A little bit late, but the new major release of version 3 has implemented Tween Instances which is exposing its play state.

```cs
public class Example : MonoBehaviour {
  readonly PositionXTween tween = new() {
    to = 1000,
    duration = 1,
  };
  TweenInstance instance;

  void Start() {
    instance = gameObject.AddTween(tween);
  }

  void Update() {
    Debug.Log(instance.isPaused);
  }
}
```

I hope this helps you! 🙌🏼
